### ISSUE TITLE
Import data from Excel

### ISSUE THREAD

**Issue by smith6jt-cop** [ColemanRHarris/mxnorm#10]
Hello,

Is it possible to simply import data directly from Excel?  I am getting a 

Error in object[[i]] : object of type 'closure' is not subsettable

error after attempting to create S3 object.

**Comment by ColemanRHarris** [comment]
Hi @smith6jt-cop, thank you for your comment. Can you share a larger code sample of what sort of import process you're trying to run? Thanks.

**Comment by smith6jt-cop** [comment]
Sure, 
First off, this is what my data look like:
![image](https://github.com/ColemanRHarris/mxnorm/assets/73438963/6246c2f1-054b-423d-89b3-0c2d6525252f)
Then in RStudio, I click the "Import Dataset" button to import from a .xlsx file.
I have successfully installed the program as I can load the sample data and normalize it.
For my data I run:

`mx_data = mx_dataset(data=X20-06-cc2C-Reg1_test,
                     slide_id="slide_id",
                     image_id="image_id",
                     marker_cols=c("CD8","CD20","Ki67","CD3e","CD68","CD21","CD4","CD45RO","CD107a","HLA-DR","FoxP3","CD163","CD15","CD45"),
                     metadata_cols=NULL)`

where X20-06-cc2C-Reg1_test is the name of my datafile after importing.

**Comment by smith6jt-cop** [comment]
[20-06-cc2C-Reg1_test.csv](https://github.com/ColemanRHarris/mxnorm/files/12818988/20-06-cc2C-Reg1_test.csv)

I have changed methods for uploading the attached .csv using:

 df = read.csv("C:/Users/smith6jt/Desktop/LN_Spleen_QuPath/Spleen_MoreMarkers/Output/20-06-cc2C-Reg1/20-06-cc2C-Reg1_test.csv")

I then run:

mx_data = mx_dataset(data=df,
                     slide_id="slide_id",
                     image_id="image_id",
                     marker_cols=c(`CD8: Cell: Mean`,`CD20: Cell: Mean`,`Ki67: Cell: Mean`,`CD3e: Cell: Mean`,`CD68: Cell: Mean`,`CD21: 
Cell: Mean`,`CD4: Cell: Mean`,`CD45RO: Cell: Mean`,`CD107a: Cell: Mean`,`HLA-DR: Cell: Mean`,`FoxP3: Cell: Mean`,`CD163: Cell: Mean`,`CD15: Cell: Mean`,`CD45: Cell: Mean`),
                     metadata_cols=c(`Signed distance to annotation Blood_Vessel`, `Signed distance to annotation Sinusoid`))

and summary(mx_dataset) still results in:

Error in object[[i]] : object of type 'closure' is not subsettable

After posting this is looks like the back ticks aren't visible around the column names, but they are there to deal with spaces.

**Comment by ColemanRHarris** [comment]
Hi @smith6jt-cop, I believe this is an import error. If you wish to keep the original column names in your dataset, please use the following commands:
```
## read in data frame with original columns
df = read.csv("C:/Users/smith6jt/Desktop/LN_Spleen_QuPath/Spleen_MoreMarkers/Output/20-06-cc2C-Reg1/20-06-cc2C-Reg1_test.csv",check.names = FALSE)

## build mx_dataset
mx_data=mx_dataset(df,slide_id = "slide_id",image_id = "image_id",
                    marker_cols = c("CD8: Cell: Mean",
                                    "CD20: Cell: Mean",
                                    "Ki67: Cell: Mean",
                                    "CD3e: Cell: Mean",
                                    "CD68: Cell: Mean",
                                    "CD21: Cell: Mean",
                                    "CD4: Cell: Mean",
                                    "CD45RO: Cell: Mean",
                                    "CD107a: Cell: Mean",
                                    "HLA-DR: Cell: Mean",
                                    "FoxP3: Cell: Mean",
                                    "CD163: Cell: Mean",
                                    "CD15: Cell: Mean",
                                    "CD45: Cell: Mean"),
                   metadata_cols = c("Signed distance to annotation Blood_Vessel",
                                     "Signed distance to annotation Sinusoid"))
```
I did also notice there is only one slide ID and one image ID for the full dataset – I'm not sure if this is intended, but `mxnorm` will likely not work as expected with only one slide and one image. Please let me know if this fixes your problems!

**Comment by smith6jt-cop** [comment]
Thanks for the info.  Yes I was testing with one slide and one image to troubleshoot the import.  I added another slide and image to the csv for further testing.

I noticed that when importing the csv with my original method, the data had three object types (tble_df, tbl, and data.frame), but when using read.csv only data.frame.

Using read.csv I still had a lot of problems with my column names.  It would be helpful to not have to rename them all as these are standard QuPath names.  What worked after several "All column identifiers must be in the dataset" and "not found" errors was taking the "-" out of "HLA-DR" so the column was just "HLADR".  I also just changed the metadata_cols to "A" and "B" as Signed distance to annotation Blood_Vessel could not be found, probably because of the underscore.  Then everything worked with only quotation marks.

Thank you for this great software!
